omod/src/main/java/org/openmrs/module/smartonfhir/web/servlet/SmartLaunchOptionSelected.java

```diff
@@ -38,9 +41,18 @@ public void doGet(HttpServletRequest req, HttpServletResponse res) throws IOExce
 		String patientId = getParameter(req, "patientId");
 		String visitId = getParameter(req, "visitId");
 		String decodedUrl = URLDecoder.decode(token, StandardCharsets.UTF_8.name());
-		String launchType = getParameterFromStringUrl(decodedUrl, "launchType");
 		
-		if (launchType.equals("encounter") && visitId == null) {
+		String jwtKeyToken = null;
+		try {
+			jwtKeyToken = getParameterFromStringUrl(decodedUrl, "key");
+		}
+		catch (URISyntaxException e) {
+			e.printStackTrace();
+		}
+		
+		String launchTypeString = getLaunchTypeString(jwtKeyToken);
```

Comment:
If `launchTypeString` comes back as `null`, we should just abort the request and return something like a 403 error here.

Reply:
@ibacher thank I will change this.